Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -13,84 +13,141 @@ References:
+            SecurityEvent
+            | extend ActorUsername = strcat(SubjectDomainName, @'\', SubjectUserName)
+                , ActorDomainName = SubjectDomainName
+                , ActingProcessId = tostring(toint(tolong(ProcessId)))
+                , RegistryKey = iif(ObjectName startswith @"\REGISTRY\MACHINE", replace_string(ObjectName, @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                , replace_string(ObjectName, @"\REGISTRY\USER","HKEY_USERS"))
+                , Type = "SecurityEvent"
+            | project-rename ActorUserId = SubjectUserSid
+                , ActorSessionId = SubjectLogonId
+                , ActingProcessName = ProcessName
+        };
+        let EventTypeLookup = datatable (AccessMask:string,EventType:string)
+        [
+            "0x1", "RegistryValueRead"
+            , "0x10", "MetadataModified"
+            , "0x10000", "ObjectDeleted"
+            , "0x2", "RegistryValueSet"
+            , "0x20000", "MetadataAccessed"
+            , "0x20006", "MetadataAccessed"
+            , "0x40000", "MetadataModified"
+            , "0x8", "RegistrySubkeyEnumerated"
+            , "0x80000", "MetadataModified"
+        ];
+        let RegistryAction = datatable (EventOriginalSubType:string,EventType:string)
+        [
+            "%%1904", "RegistryValueSet"
+            , "%%1905", "RegistryValueSet"
+            , "%%1906", "RegistryValueDeleted"
+        ];
+        let RegistryType = datatable (TypeCode:string,TypeName:string)
+        [
+            "%%1872", "REG_NONE"
+            , "%%1873",	"REG_SZ"
+            , "%%1874", "REG_EXPAND_SZ"
+            , "%%1875", "REG_BINARY"
+            , "%%1876", "REG_DWORD"
+            , "%%1879", "REG_MULTI_SZ"
+            , "%%1883", "REG_QWORD"
+        ];
+        union isfuzzy=false (
+            WindowsEvent
+            | where EventID == 4663
+                and EventData.ObjectType == "Key"
+            | extend AccessMask = tostring(EventData.AccessMask)
+                , Type = "WindowsEvent"
+            | lookup EventTypeLookup on AccessMask
+            | invoke ASIM_ParseWindowsEvents()
+            | project TimeGenerated, Computer, EventID, EventType, ActorUsername, ActorDomainName, ActorUserId, ActorSessionId, ActingProcessName, ActingProcessId, RegistryKey, _ResourceId, Type
+        ), (
+            WindowsEvent
+            | where EventID == 4657
+            | invoke ASIM_ParseWindowsEvents()
+            | extend EventOriginalSubType = tostring(EventData.OperationType)
+                , OldValue = tostring(EventData.OldValue)
+                , NewValue = tostring(EventData.NewValue)
+                , RegistryValue = tostring(EventData.ObjectValueName)
+                , NewValueType = tostring(EventData.NewValueType)
+                , OldValueType = tostring(EventData.OldValueType)
+            | lookup RegistryAction on EventOriginalSubType
```

Comment:
Purely style, but post this line, both WindowsEvent and SecurityEvent for 4657 are the same, so you could union just those two and use just one copy.

Reply:
Done